job-service/src/main/java/com/hpe/caf/services/job/jobtype/JobType.java

```diff
@@ -81,21 +64,15 @@ public String getId() {
      */
     public WorkerAction buildTask(
         final String partitionId, final String jobId, final JsonNode parameters
-    ) throws BadRequestException, InvalidJobTypeDefinitionException {
-        final WorkerAction task = new WorkerAction();
-        task.setTaskClassifier(taskClassifier);
-        task.setTaskApiVersion(taskApiVersion);
-        task.setTaskPipe(taskPipe);
-        task.setTargetPipe(targetPipe);
-        final JsonNode taskData = taskDataBuilder.build(partitionId, jobId, parameters);
-        // job-put expects `Map` (or `String`), but only does a shallow type check
+    ) throws BadRequestException, InvalidJobTypeDefinitionException
+    {
+        final JsonNode task = taskBuilder.build(partitionId, jobId, parameters);
+        JsonSchemaTaskScriptValidator.getInstance().validate(task);
```

Comment:
Wasn't sure about this static `JsonSchemaTaskScriptValidator` class here, but couldn't really find a nicer way of doing this, due to these classes are initialized on startup.